src/connection.h

```diff
@@ -8,21 +10,7 @@
 #include <cstring>
 
 SEXP read_bin(SEXP con, size_t bytes = 64 * 1024);
-SEXP write_bin(SEXP data, SEXP con);
-
-inline SEXP R_GetConnection(SEXP con) { return con; }
-
-inline size_t R_ReadConnection(SEXP con, void* buf, size_t n) {
-  SEXP res = PROTECT(read_bin(con, n));
-
-  R_xlen_t size = Rf_xlength(res);
-
-  memcpy(buf, RAW(res), size);
-
-  UNPROTECT(1);
-
-  return Rf_xlength(res);
-}
+cpp11::sexp write_bin(cpp11::sexp data, cpp11::sexp con);
 
 inline size_t R_WriteConnection(SEXP con, void* buf, size_t n) {
```

Comment:
This uses `memcpy()` and I wasn't sure how to do this properly with cpp11.

Reply:
I don't think there is an alternative to `memcpy()`, but you could allocate a cpp11 `raws` and then do `payload.sexp()` to get access to the SEXP you can call `RAW()` on. I don't think that is too awful, and at least lets cpp11 manage the memory